src/main/java/graphql/util/Traverser.java

```diff
@@ -0,0 +1,453 @@
+/*
+ * Generic traverser 
+ * 
+ * Created: Jan 28, 2018 12:42:17 PM
+ * Author: gkesler
+ */
+package graphql.util;
+
+import static graphql.Assert.assertShouldNeverHappen;
+import java.util.ArrayDeque;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ *
+ * @author gkesler
+ */
+public class Traverser<T> {
```

Comment:
could you please break this class up in several files? each interface etc should live in its own file. maybe its own package if you think they should be recognizes as belonging together.

Reply:
sure, will do